### PR TITLE
tracer: Add WithStackFrames start option

### PR DESCRIPTION
**Pull request by matoous**
Add WithStackFrames start option that allows setting the number of
stack frames taken and stack frames skipped globally for the tracer.

Fixes: https://github.com/DataDog/dd-trace-go/issues/443
Signed-off-by: Matous Dzivjak <matous.dzivjak@kiwi.com>

### CODE REVIEW

**Comment by cgilmour** [comment]
Could you rebase this to resolve the conflicts with the `v1` branch? Thanks

**Comment by matoous** [comment]
@cgilmour done. The CI fails but with "failed to fetch dependencies" so I hope I didn't cause it.

**Comment by cgilmour** [comment]
The CI failure is from #448, we’ll resolve that soon.

File: ddtrace/tracer/span.go [hunk 1]
```diff
@@ -57,8 +57,10 @@ type span struct {
 	ParentID uint64             `msg:"parent_id"`         // identifier of the span's direct parent
 	Error    int32              `msg:"error"`             // error status of the span; 0 means no errors
 
-	finished bool         `msg:"-"` // true if the span has been submitted to a tracer.
-	context  *spanContext `msg:"-"` // span propagation context
+	finished    bool         `msg:"-"` // true if the span has been submitted to a tracer.
+	context     *spanContext `msg:"-"` // span propagation context
+	stackFrames uint         `msg:"-"` // number of stack frames
```
Comment (gbbr) [inline ddtrace/tracer/span.go]: I'm not sure about having these here. I'm also unsure about suggesting a good alternative. Maybe having `*config` (the tracer's config) here makes more sense.

Comment (matoous) [reply ddtrace/tracer/span.go]: Please take a look and tell me if it is better now.

**Comment by gbbr** [comment]
@matoous sorry for taking so long to respond on this. Seeing the implementation now, I'm not sure that this is something we want to add now. It seems a bit harsh to set limits like this throughout the code, globally. I'm thinking the finish option that you've added in #436 should be good enough for now, but happy to discuss further.

If you don't mind, let's leave this on hold for a while. What do you think?

**Comment by gbbr** [comment]
I'm going to be closing this for now, but happy to pick it up again if you want to talk about it.

**Pull request closed without merging**